src/core/contracts/identity.ts

```diff
@@ -183,6 +183,185 @@ const getBeaconAddress = async (opts?: ConfigOpts): Promise<EthereumAddress> =>
   return address;
 };
 
+/**
+ * DelegateAdd represents a struct for adding delegates
+ */
+interface DelegateAdd extends TypedMessageDataField {
+  nonce: number;
+  delegateAddr: HexString;
+  role: DelegationRole;
+}
+
+export interface DelegateAddParams {
+  nonce?: number;
+  delegateAddr: HexString;
+  role: DelegationRole;
+}
+
+/**
+ * DomainData represents EIP-712 unique domain
+ */
+export type TypedDomainData = {
+  name: string;
+  version: string;
+  chainId: number;
+  verifyingContract: string;
+  salt: string;
+};
+
+/**
+ * TypedData represents EIP-712 complete typed data
+ */
+export interface TypedData {
+  types: Record<string, Array<TypedDataField>>;
+  primaryType: string;
+  domain: TypedDomainData;
+  message: Record<string, unknown>;
+}
+
+type TypedMessageDataField = Record<string, unknown>;
+
+/**
+ * EIP712Signature represents a ECDSA r value, s value and EIP-155 calculated Signature v value
+ */
+export type EIP712Signature = { r: string; s: string; v: number };
+
+/**
+ * TypedDataField represents a single type
+ */
+export interface TypedDataField {
+  name: string;
+  type: string;
+}
+
+const EIP712_DOMAIN_TYPES: TypedDataField[] = [
+  { name: "name", type: "string" },
+  { name: "version", type: "string" },
+  { name: "chainId", type: "uint256" },
+  { name: "verifyingContract", type: "address" },
+  { name: "salt", type: "bytes32" },
+];
```

Comment:
Not blocking, but I wonder if we should pull all these EIP712 types out to a separate file.